### ISSUE TITLE
feature request: displaying subtitles

### ISSUE THREAD

**Issue by iuqiddis** [ozmartian/vidcutter#100]
I love the program. It's great at quick cuttings jobs. Is there a possibility to toggle subtitles on/off on videos that already have the embedded subtitle track, or via an external .srt file (that the user can load alongside the video file)?

Thanks.

**Comment by ozmartian** [comment]
Hey there,



It should already be doing that by default, it always attempts to maintain all audio and text streams present in the original media file. Of course, there are often instances where a media file is encoded badly in some way or subtitle metadata is invalid somehow. Basically what happens is a first attempt is made including all audio and subtitle tracks present on the original. If that fails or produces a file too small in size to be considered valid, we then make a second attempt using FFMpeg's default  where FFMpeg determines the best audio tracks re quality and usually just the very first subtitle track. So you should get a video produced with at least one of the subtitle tracks still on there after cutting ad joining. If this second attempt again fails it then reverts to a basic format of single video and audio, excluding all else like menus and subtitles.



So while there is no "subtitle" options or mention in the interface, behind the scenes there are considered. The upcoming version 4.5 further improves the way these is all handled and then for 5.0 version we will add ability for you to tweak things in greater detaill  like selecting from one or more audio , subtitle tracks. Also, if you check the apps log file it should state every time it switches betweem all streams (incl subtitles) and not so check it out (log file option in right click menu / menu button icon)



I'm more than happy to dig deeper here as you may have a video file I have not come across before and may be able to help me better support all formats. If you still cannot get any subtitles in the final video then you need to get a small sample of your original video file to me. Maybe use some other video editor that its all works with so you can just cut a sample of 20-30 secs and upload it someplace and email me the link.



Cheers,

Pete




---- On Tue, 24 Oct 2017 07:50:47 +0800 iuqiddis &lt;notifications@github.com&gt; wrote ----




I love the program. It's great at quick cuttings jobs. Is there a possibility to toggle subtitles on/off on videos that already have the embedded subtitle track, or via an external .srt file (that the user can load alongside the video file)?

Thanks.

**Comment by iuqiddis** [comment]
Thanks for the quick reply, and sorry for the delay on my part.

I was indeed mistaken. Going back, I realized the subtitles tracks were indeed included in the output file. I guess as a feature request, is it possible to 'burn the subtitles' into the output clips? The reason I am asking for this is that it would allow quick sharing of short clips (one to five seconds of dialogue) as 'video gifs' (mp4 or webm files) over the various messaging apps instead of lower quality gifs; their built-in players don't seem to support subtitles track as far as i can tell (I've tried whatsapp and viber).

Of course this request may be totally out of the scope of the program, which is all perfectly understandable. cheers, and thanks again.
